js/form.js

```diff
@@ -0,0 +1,80 @@
+class searchForm {
+  constructor(formDiv) {
+    this.formDiv = formDiv;
+    this.#build();
+    this.loader = document.getElementById("loader");
+  }
+
+  #build() {
+    this.form = document.createElement("form");
+    Object.assign(this.form, {
+      class: "d-flex",
+      role: "search",
+    });
+    this.form.setAttribute("class", "d-flex");
+    this.input = document.createElement("input");
+    Object.assign(this.input, {
+      class: "form-control me-2",
+      type: "search",
+      placeholder: "Search",
+      id: "searchInput",
+    });
+    this.input.setAttribute("class", "form-control me-2");
+    this.button = document.createElement("button");
+    Object.assign(this.button, {
+      class: "btn btn-outline-success",
+      type: "submit",
+      id: "searchButton",
+      innerHTML: "Search",
+    });
+    this.span = document.createElement("span");
+    Object.assign(this.span, {
+      class: "loader",
+      id: "loader",
+    });
+    this.span.setAttribute("class", "loader");
+    this.button.setAttribute("class", "btn btn-outline-success");
+    this.form.appendChild(this.input);
+    this.form.appendChild(this.button);
+    this.formDiv.appendChild(this.form);
+    this.formDiv.appendChild(this.span);
+
+    this.button.addEventListener("click", async (e) => {
+      e.preventDefault();
+      this.loader.setAttribute("style", "display:block;");
+      let companies = await this.searchCompany();
+      this.renderResult(companies);
+      this.loader.setAttribute("style", "display:none;");
+    });
+
+    this.input.addEventListener("input", (e) => {
+      if (this.input.value != "") {
+        this.loader.setAttribute("style", "display:block;");
+      } else {
+        this.loader.setAttribute("style", "display:none;");
+      }
+    });
+  }
+  async getApi(url) {
```

Comment:
this function is written more then once in your code. why? you can declere it once and then use it. does it have to be a method inside a class at all? think about DRY principle.